error-issue/test/error_monitor.test.ts

```diff
@@ -0,0 +1,208 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ErrorMonitor} from '../src/error_monitor';
+import {StackdriverApi} from '../src/stackdriver_api';
+import {Stackdriver} from '../src/types';
+
+import nock from 'nock';
+import {mocked} from 'ts-jest/utils';
+
+describe('ErrorMonitor', () => {
+  let monitor: ErrorMonitor;
+  const stackdriver = ({
+    listGroups: jest.fn(),
+    setGroupIssue: jest.fn(),
+  } as unknown) as StackdriverApi;
+
+  const infrequentGroup: Stackdriver.ErrorGroupStats = {
+    group: {
+      name: 'Error: Infrequent error',
+      groupId: 'infrequent_id',
+    },
+    count: 2000,
+    timedCounts: [
+      {
+        count: 4,
+        startTime: new Date('Feb 25, 2020'),
+        endTime: new Date('Feb 26, 2020'),
+      },
+    ],
+    firstSeenTime: new Date('Feb 20, 2020'),
+    representative: {message: 'Error: Infrequent error'},
+  };
+  const acknowledgedGroup: Stackdriver.ErrorGroupStats = {
+    group: {
+      name: 'Error: Acknowledged error',
+      groupId: 'acknowledged_id',
+      trackingIssues: [{url: 'https://github.com/blah/blah'}],
+    },
+    count: 20000,
+    timedCounts: [
+      {
+        count: 6000,
+        startTime: new Date('Feb 25, 2020'),
+        endTime: new Date('Feb 26, 2020'),
+      },
+    ],
+    firstSeenTime: new Date('Feb 20, 2020'),
+    representative: {message: 'Error: Acknowledged error'},
+  };
+  const newGroup: Stackdriver.ErrorGroupStats = {
+    group: {
+      name: 'Error: New error',
+      groupId: 'new_id',
+    },
+    count: 20000,
+    timedCounts: [
+      {
+        count: 6000,
+        startTime: new Date('Feb 25, 2020'),
+        endTime: new Date('Feb 26, 2020'),
+      },
+    ],
+    firstSeenTime: new Date('Feb 20, 2020'),
+    representative: {message: 'Error: New error'},
+  };
+  const errorGroups = [infrequentGroup, acknowledgedGroup, newGroup];
+  const newReport = {
+    errorId: 'new_id',
+    firstSeen: new Date('Feb 20, 2020'),
+    dailyOccurrences: 6000,
+    stacktrace: 'Error: New error',
+  };
+
+  beforeAll(() => {
+    nock.disableNetConnect();
+  });
+
+  afterAll(() => {
+    nock.enableNetConnect();
+  });
+
+  beforeEach(() => {
+    monitor = new ErrorMonitor(stackdriver);
+    mocked(stackdriver.listGroups).mockResolvedValue(errorGroups);
+    nock.cleanAll();
+  });
+
+  afterEach(() => {
+    // Fail the test if there were unused nocks.
+    if (!nock.isDone()) {
+      throw new Error('Not all nock interceptors were used!');
+    }
+    nock.cleanAll();
+  });
+
+  describe('newErrorsToReport', () => {
+    it('ignores infrequent errors', async () => {
```

Comment:
I think it's better to merge these three tests by having an `expect(newErrorIds).toMatchObject(['acknowledged_id', 'new_id']);`
The printout will display a readable diff if the expectation isn't met

Reply:
I usually lean toward splitting in a case like this for the "tests as documentation" reason, but I've merged them here